src/core/src/parallel/thread_local_store.rs

```diff
@@ -0,0 +1,100 @@
+use std::cell::RefCell;
+use std::ops::{AddAssign, Deref};
+
+use thread_local::CachedThreadLocal;
+
+/// A simple struct that delivers thread local variables.
+///
+/// Some computations need thread local copies of the memory
+/// to avoid data races. This is a simple wrapper over
+/// the `thread_local` crate that exposes a simple API
```

Comment:
That's a lot of simple in the same sentence =). Maybe we should also add the fact that this `Deref<Target=RefCell>` here?